packages/pipeline/src/utils/get_ohlcv_trading_pairs.ts

```diff
@@ -0,0 +1,100 @@
+import axios from 'axios';
+import { chain } from 'ramda';
+import { Connection } from 'typeorm';
+
+export interface TradingPair {
+    fromSymbol: string;
+    toSymbol: string;
+    latest: number;
+}
+
+interface StaticPair {
+    fromSymbol: string;
+    toSymbol: string;
+}
+
+const COINLIST_API = 'https://min-api.cryptocompare.com/data/all/coinlist?BuiltOn=7605';
+
+interface CryptoCompareCoinListResp {
+    Data: Map<string, CryptoCompareCoin>;
+}
+
+interface CryptoCompareCoin {
+    Symbol: string;
+    BuiltOn: string;
+    SmartContractAddress: string;
+}
+
+const TO_CURRENCIES = ['USD', 'EUR', 'ETH'];
+
+/**
+ * Get trading pairs with latest scraped time for OHLCV records
+ * @param conn a typeorm Connection to postgres
+ */
+export async function fetchOHLCVTradingPairs(
+    conn: Connection,
+    source: string,
+    earliestBackfillTime: number,
+): Promise<TradingPair[]> {
+    const rawTokenAddresses: Array<{ tokenaddress: string }> = await conn.query(
+        `SELECT DISTINCT(maker_token_address) as tokenaddress FROM raw.exchange_fill_events UNION
+        SELECT DISTINCT(taker_token_address) as tokenaddress FROM raw.exchange_fill_events
+        LIMIT 1`,
+    );
+    const tokenAddresses = rawTokenAddresses.map(obj => obj.tokenaddress);
+
+    // get token symbols used by Crypto Compare
+    const allCoins = await axios.get<CryptoCompareCoinListResp>(COINLIST_API);
+    const erc20Coins: Map<string, string> = new Map();
+    Object.entries(allCoins.data.Data).forEach(pair => {
+        const [k, v] = pair;
+        if (v.BuiltOn === '7605' && v.SmartContractAddress !== 'N/A') {
+            erc20Coins.set(v.SmartContractAddress, k);
+        }
+    });
+
+    const mapFn = (tokenAddress: string): StaticPair[] => {
+        return chain(fiat => {
+            const fromSymbol = erc20Coins.get(tokenAddress);
+            if (!!fromSymbol) {
+                return [
+                    {
+                        toSymbol: fiat,
+                        fromSymbol,
+                    },
+                ];
+            } else {
+                return [];
+            }
+        }, TO_CURRENCIES);
+    };
+    const tradingPairs: StaticPair[] = chain(mapFn, tokenAddresses);
+
+    const tradingPairsLatestQuery: string = tradingPairs
+        .map(pair => {
+            return `SELECT
+              case COUNT(*) when 0 then ${earliestBackfillTime} else MAX(end_time) end AS latest,
+              '${pair.fromSymbol}' as from_symbol,
+              '${pair.toSymbol}' as to_symbol
+            FROM raw.ohlcv_external
+            WHERE
+              from_symbol = '${pair.fromSymbol}' AND
+              to_symbol = '${pair.toSymbol}' AND
+              source = '${source}'`;
+        })
+        .join('\nUNION\n');
+
+    const latestTradingPairsResult: Array<{
+        from_symbol: string;
+        to_symbol: string;
+        latest: string;
+    }> = await conn.query(tradingPairsLatestQuery);
```

Comment:
Any chance we can wrap `conn.query` so that we can append `Async` to the method name? It's a really nasty bug to forget to `await` an async method, and also really easy to make that mistake. It really helps to have the suffix convention that we do, but for external libraries without the convention, it's worth wrapping them to keep our codebase consistent.

Reply:
@fabioberger this doesn't seem to be a convention that we follow throughout our codebase. It seems like forgetting to add `await` is about as likely as forgetting to wrap the function and append `Async`, so I'm not sure what this buys us.

Instead I would propose we improve our linter rules to do either one of the following:

1. If a function returns a promise, you must call it `await`. This is what we want to do 99% of the time and you can always disable the linter rule in the rare case where you intended to call the function without `await`. There may even already be a rule that we could copy/use.
2. If the first option is not possible, we can add a linter rule to enforce wrapping of external functions with an `Async` prefix.

What do you think?

Reply:
@albrow let me backtrack a bit.

We already have the `no-floating-promises` which should watch missing awaits within our codebase. We didn't always have this TSLint rule which is why we had the `async` prefix convention across our codebase. We do however want to keep the `async` prefix convention for our publicly exposed interface, since the users of our libraries might not use a linter and this way it's very easy for them to immediately tell which methods return promises and which don't. 

In this specific case then, we can leave it as it is.